crates/matrix-sdk-base/src/client.rs

```diff
@@ -519,7 +519,19 @@ impl BaseClient {
         };
 
         #[cfg(not(feature = "e2e-encryption"))]
-        let to_device = response.to_device.events;
+        let to_device = response
+            .to_device
+            .events
+            .into_iter()
+            .filter(|raw| {
+                if let Ok(Some(event_type)) = raw.get_field::<String>("type") {
+                    event_type != "m.room.encrypted"
+                } else {
+                    false // Exclude events with no type or encrypted
+                }
+            })
+            .map(matrix_sdk_common::deserialized_responses::ProcessedToDeviceEvent::PlainText)
```

Comment:
When there is no encryption, to-device are not processed by the olm machine. There is no `UnProcessed` variant of `ProcessedToDeviceEvent` (not sure it would make sense :/).
So anyhow, we can just ignore the encrypted to-device as they are useless. So we can just map to PlainText variant?
I wonder if we should at some point remove the optional `e2e-encryption` feature flag